packages/vitest/src/node/cli/cac.ts

```diff
@@ -0,0 +1,237 @@
+      .split('\n')
+      .map(line => line.startsWith(subcommandMarker) ? line.split(subcommandMarker)[1] : '')
+      .filter(line => line.length !== 0)
+      .join('\n')
+
+    if (!options.body) {
+      addBannerWarning('no options were found for your subcommands so we printed the whole output')
+      options.body = defaultOutput
+    }
+
+    return info
+  })
+
+  cli
+    .command('run [...filters]')
+    .action(run)
+
+  cli
+    .command('related [...filters]')
+    .action(runRelated)
+
+  cli
+    .command('watch [...filters]')
+    .action(watch)
+
+  cli
+    .command('dev [...filters]')
+    .action(watch)
+
+  cli
+    .command('bench [...filters]')
+    .action(benchmark)
+
+  // TODO: remove in Vitest 2.0
+  cli
+    .command('typecheck [...filters]')
+    .action(() => {
+      throw new Error(`Running typecheck via "typecheck" command is removed. Please use "--typecheck" to run your regular tests alongside typechecking, or "--typecheck.only" to run only typecheck tests.`)
+    })
+
+  cli
+    .command('[...filters]')
+    .action((filters, options) => start('test', filters, options))
+
+  return cli
+}
+
+async function runRelated(relatedFiles: string[] | string, argv: CliOptions): Promise<void> {
+  argv.related = relatedFiles
+  argv.passWithNoTests ??= true
+  await start('test', [], argv)
+}
+
+async function watch(cliFilters: string[], options: CliOptions): Promise<void> {
+  options.watch = true
+  await start('test', cliFilters, options)
+}
+
+async function run(cliFilters: string[], options: CliOptions): Promise<void> {
+  options.run = true
+  await start('test', cliFilters, options)
+}
+
+async function benchmark(cliFilters: string[], options: CliOptions): Promise<void> {
+  console.warn(c.yellow('Benchmarking is an experimental feature.\nBreaking changes might not follow SemVer, please pin Vitest\'s version when using it.'))
+  await start('benchmark', cliFilters, options)
+}
+
+function normalizeCliOptions(argv: CliOptions): CliOptions {
+  if (argv.root)
+    argv.root = normalize(argv.root)
+  else
+    delete argv.root
+
+  if (argv.config)
+    argv.config = normalize(argv.config)
+  else
+    delete argv.config
+
+  if (argv.workspace)
+    argv.workspace = normalize(argv.workspace)
+  else
+    delete argv.workspace
+
+  if (argv.dir)
+    argv.dir = normalize(argv.dir)
+  else
+    delete argv.dir
+
+  if (argv.exclude) {
+    argv.cliExclude = toArray(argv.exclude)
+    delete argv.exclude
+  }
+
+  if (argv.coverage) {
+    const coverage = argv.coverage
+    if (coverage.exclude)
+      coverage.exclude = toArray(coverage.exclude)
+
+    if ((coverage as BaseCoverageOptions).include)
+      (coverage as BaseCoverageOptions).include = toArray((coverage as BaseCoverageOptions).include)
+
+    if ((coverage as CoverageIstanbulOptions).ignoreClassMethods)
+      (coverage as CoverageIstanbulOptions).ignoreClassMethods = toArray((coverage as CoverageIstanbulOptions).ignoreClassMethods)
+  }
```

Comment:
Do we still need these as there's `CLIOption.array` flag? Would it automatically convert these?

Reply:
Are these `enabled` setters still needed too?

https://github.com/vitest-dev/vitest/blob/dcf0d6afe974487a53f7e7c90c0f79d29fa885d2/packages/vitest/src/node/cli-api.ts#L40-L62